File: java/demo/src/main/java/cn/huidu/device/demo/demo1_deviceInfod/GetDeviceInfo.java
package cn.huidu.device.demo.demo1_deviceInfod;

import cn.huidu.device.sdk.Device;

public class GetDeviceInfo {
    public static void main(String[] args) {
        // Replace with your own parameters
        // String host = "${YourHost}";
        // String sdkKey = "${YourSdkKey}";
        // String sdkSecret = "${YourSdkSecret}";
        // Config.InitSdk(host, sdkKey, sdkSecret);

        // 1. Instantiate device interface object
        Device device = new Device();
        // 2. Call interface to get information (device properties) for specified device ("C16-D21-015BD", need to replace with your own device id)
        String jsonResult = device.getDeviceProperty("");
        // 3. Print result
        System.out.println(jsonResult);
    }

}

File: python/demos/demo1_deviceInfo/GetDeviceInfo.py
from sdk.common.Config import Config
from sdk.Device import Device
import json

def main():
    # with your own parameters
    # host = "${YourHost}"
    # sdk_key = "${YourSdkKey}"
    # sdk_secret = "${YourSdkSecret}"
    # Config.init_sdk(host, sdk_key, sdk_secret)

    # Instantiate a device interface object
    device = Device()

    # Call the interface to get the online device
    result = device.get_device_property(ids="C16L-D00-A000F")

    # Print the result
    print("API Respond to the results:")
    print(json.dumps(result, indent=2))

if __name__ == "__main__":
    main()
